### PR TITLE
allows you to specify 'strategy'

### PR DESCRIPTION
**Pull request by orthagonal**
added test case that demonstrates this

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -33,6 +33,32 @@ exports.register = (server, pluginOptions, next) => {
...
+        }
+      ]
+    ]
+   }
+  */
+  if (pluginOptions.strategy) {
+    const options = {
+      apiKeys: {}
+    };
+    pluginOptions.strategy.apiKeys.forEach((apikey) => {
```
Comment (jgallen23) [inline index.js]: what is this for?

Comment (jgallen23) [inline index.js]: this apikeys an array or an object?

Comment (jgallen23) [inline index.js]: can we make it an object instead?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hapi-api-key",
-  "version": "1.0.0",
+  "version": "1.1.0",
```
Comment (jgallen23) [inline package.json]: can you cahnge this back to 1.0.0?

File: index.js (hunk 1 above)
Comment (jgallen23) [inline index.js]: ping

Comment (orthagonal) [inline index.js]: I changed it to be an object.

File: package.json (hunk 2 above)
Comment (orthagonal) [inline package.json]: This is done as well.

File: index.js (hunk 1 above)
Comment (jgallen23) [inline index.js]: no, it's an array....

Comment (orthagonal) [inline index.js]: Right now we pass the api keys for a given strategy as an array of objects like so:

strategy: {
    name: 'myStrategyName',
    mode: true // (can be any valid strategy mode)
    apiKeys: [
      {
        key: 'anAPIKey',
        credentials: {
          name: 'authenticationName'
        }
    ]
   }

  I'd rather apiKeys just be an object, with each key being the api key and the value being the credentials object.  But confi (or someone) doesn't seem to be evaluating the handlebars expressions when they are in the keys of yaml.  If we can get around that obstacle we can have apiKeys just be a simple object.

**Pull request merged**